sbpy/bib/core.py

```diff
@@ -29,8 +29,26 @@ def register(task, citations):
     """
     global _bibliography, _track
     if _track:
-        _bibliography[task] = _bibliography.get(task, {})
-        _bibliography[task].update(citations)
+        if task in _bibliography:
+            key = []
+            value = set()
+            key.append(list(citations.keys()))
+            for i in range(len(key)):
```

Comment:
instead of working with indices, you can get the keys and values directly using something like:
`for newsubtask, newcitation in citation.items():`